sdk/storage/azure-storage-blobs/src/block_blob_client.cpp

```diff
@@ -219,8 +225,13 @@ namespace Azure { namespace Storage { namespace Blobs {
     _internal::FileReader fileReader(fileName);
 
     auto uploadBlockFunc = [&](int64_t offset, int64_t length, int64_t chunkId, int64_t numChunks) {
+#if defined(AZ_PLATFORM_WINDOWS)
+      Azure::Core::IO::_internal::RandomAccessFileBodyStream contentStream(
+          reinterpret_cast<void*>(fileReader.GetHandle()), offset, length);
```

Comment:
I'll fix this in a separate PR.

`fileReader.GetHandle()` will return `void*` directly, so we don't need `if defined` here.

Reply:
I was having trouble making this work within the lambda, it was easier for me to just fix it in this PR. Already done, please review :)